backend/tracim_backend/views/core_api/user_controller.py

```diff
@@ -986,6 +1005,214 @@ def about_user(self, context, request: TracimRequest, hapic_data: HapicData) ->
+    @hapic.input_path(UserPicturePathSchema())
+    @hapic.handle_exception(UnavailablePreview, HTTPStatus.BAD_REQUEST)
+    @hapic.handle_exception(PageOfPreviewNotFound, HTTPStatus.BAD_REQUEST)
+    @hapic.handle_exception(PreviewDimNotAllowed, HTTPStatus.BAD_REQUEST)
+    @hapic.handle_exception(UserAvatarNotFound, HTTPStatus.BAD_REQUEST)
+    @hapic.output_file([])
+    def get_preview_avatar(
+        self, context, request: TracimRequest, hapic_data: HapicData
+    ) -> HapicFile:
+        width = DEFAULT_AVATAR_SIZE.width
+        height = DEFAULT_AVATAR_SIZE.height
+        user_api = UserApi(
+            current_user=request.current_user, session=request.dbsession, config=request.app_config
+        )
+        default_filename = "avatar.jpg".format(width=width, height=height)
+        return user_api.get_avatar_preview(
+            request.candidate_user.user_id,
+            filename=hapic_data.path.filename,
+            default_filename=default_filename,
+            width=width,
+            height=height,
+            force_download=hapic_data.query.force_download,
+        )
+
+    @hapic.with_api_doc(tags=[SWAGGER_TAG__USER_ENDPOINTS])
+    @check_right(knows_candidate_user)
+    @hapic.input_query(FileQuerySchema())
+    @hapic.input_path(UserPicturePathSchema())
+    @hapic.handle_exception(UserAvatarNotFound, HTTPStatus.BAD_REQUEST)
+    @hapic.output_file([])
+    def get_raw_avatar(self, context, request: TracimRequest, hapic_data: HapicData) -> HapicFile:
+        try:
+            user_api = UserApi(
+                current_user=request.current_user,
+                session=request.dbsession,
+                config=request.app_config,
+            )
+            default_filename = "avatar.jpg"
+            return user_api.get_avatar(
+                request.candidate_user.user_id,
+                filename=hapic_data.path.filename,
+                default_filename=default_filename,
+                force_download=hapic_data.query.force_download,
+            )
+        except CannotGetDepotFileDepotCorrupted as exc:
+            raise TracimFileNotFound(
+                "avatar for user {} not found in depot.".format(request.candidate_user.user_id)
+            ) from exc
+
+    @hapic.with_api_doc(tags=[SWAGGER_TAG__USER_ENDPOINTS])
+    @check_right(has_personal_access)
+    @hapic.handle_exception(NoFileValidationError, HTTPStatus.BAD_REQUEST)
+    @hapic.handle_exception(MimetypeNotAllowed, HTTPStatus.BAD_REQUEST)
+    @hapic.input_files(SimpleFileSchema())
+    @hapic.input_path(UserPicturePathSchema())
+    @hapic.output_body(NoContentSchema(), default_http_code=HTTPStatus.NO_CONTENT)
+    def put_raw_avatar(self, context, request: TracimRequest, hapic_data: HapicData) -> None:
+        if hapic_data.files.files is None:
+            raise NoFileValidationError('No file "files" given at input, validation failed.')
+        if hapic_data.files.files.type not in ALLOWED_MIMETYPE_FOR_AVATAR:
```

Comment:
note: there is not size max nor user data space check here as https://github.com/tracim/tracim/issues/3842 say "let apache do the job here", same for cover